rosserial_client/src/ros_lib/ros/node_handle.h

```diff
@@ -209,6 +224,20 @@ class NodeHandle_ : public NodeHandleBase_
     /* while available buffer, read data */
     while (true)
     {
+      // If a timeout has been specified, check how long spinOnce has been running.
+      if (spin_timeout_ > 0)
+      {
+        // If the maximum processing timeout has been exceeded, flush
+        // the receive buffer and exit with error.
+        // This is the maximum time that can be run for without caring
+        // what state the processing is in.
+        if ((hardware_.time() - c_time) > spin_timeout_)
+        {
+          // Reset back to first mode.
+          mode_ = MODE_FIRST_FF;
+          return -2;
```

Comment:
I supposed with a timeout of 0 (the default), the behaviour is unchanged, but it would be good to use a constant here instead, and also clarify that the expectation for users of `setSpinTimeout` is that they check for this return code on `spinOnce`, and if they see it, flush the buffer.

Reply:
-2 was used for the other timeout below, I could make it something else also.

Reply:
I guess this actually works for a user even if they don't explicitly flush the queue— all it does is ensure that `spinOnce` only takes up to a certain amount of time, so if there's a burst of messages which arrive all at once, those will get processed over a _series_ of spin invocations, so long as there's the buffer space to accommodate them.

In any case, yes, please define SPIN_OK and SPIN_TIMEOUT consts and return them from this function in the appropriate spots.